### PR TITLE
Feature/dev docker setup

### PR DESCRIPTION
**Pull request by vikramjadon921995**
## Why?
Docker setup is required for contributors to run containers in the development environment.

## This change addresses the need by:
+ Created a `Dockerfile-dev` for development environment.
+ Created a `docker-compose-dev.yml` for development environment.
+ Shell script to run in the web app container when it is ready.

[delivers #162893814]

## Checklist

- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

### CODE REVIEW

File: Dockerfile-dev [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  && locale-gen \
+  && update-locale LANG=$LANG \
+  # Remove unwanted package after use
+  && apt-get purge -y curl file xz-utils build-essential locales \
+  && apt-get -y autoremove \
+  && apt-get -y clean
+
+RUN ["chmod", "+x", "./docker-dev-provision.sh"]
+
+CMD ["./docker-dev-provision.sh"]
```
Comment (ashish173) [inline Dockerfile-dev]: Please leave an empty line at the end of file. Just a general setup we follow. @vikramjadon921995 

Make this change for other files in this PR as well.

Comment (vikramjadon921995) [reply Dockerfile-dev]: Sure. Will do. I remembered one more change. Need to move these files to `config/docker/dev`. Will make that change too and let you know.

File: Dockerfile-dev [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+# Apps
+COPY apps apps
+RUN mix do local.hex --force, local.rebar --force
+RUN mix do deps.get, deps.compile
+
+# Create assets build for admin app
+RUN cd apps/admin_app/assets \
+  && yarn install \
+  && yarn deploy \
```
Comment (pkrawat1) [inline Dockerfile-dev]: @vikramjadon921995 
```
yarn deploy \
&& cd .. \
&& mix phx.digest
```

not required, as these are for prod buil

Comment (vikramjadon921995) [reply Dockerfile-dev]: Okay. Will remove them.

**Pull request merged**